ehr/app/src/main/java/org/hl7/gravity/refimpl/sdohexchange/dto/converter/PersonalCharacteristicsInfoHolderToDtoConverter.java

```diff
@@ -27,28 +28,37 @@
   public PersonalCharacteristicDto convert(T infoHolder) {
     Observation obs = infoHolder.getObservation();
     PersonalCharacteristicDto dto = new PersonalCharacteristicDto();
+    dto.setId(obs.getIdElement()
+        .getIdPart());
     //Type
-    CharacteristicCode type = CharacteristicCode.fromCode(obs.getCode()
-        .getCodingFirstRep()
-        .getCode());
-    dto.setType(type);
-    //Method + detail
-    CodeableConcept method = obs.getMethod();
-    dto.setMethod(CharacteristicMethod.fromCode(method.getCodingFirstRep()
-        .getCode()));
-    dto.setMethodDetail(method.getText());
-    //Value + detail
-    if (obs.getValue() instanceof CodeableConcept) {
-      CodeableConcept value = (CodeableConcept) obs.getValue();
-      dto.setValue(new CodingDto(value.getCodingFirstRep()
-          .getCode(), value.getCodingFirstRep()
-          .getDisplay()));
-      dto.setValueDetail(value.getText());
-    } else if (obs.hasComponent() && CharacteristicCode.ETHNICITY.equals(type)) {
-      convertEthnicity(obs, dto);
-    } else if (obs.hasComponent() && CharacteristicCode.RACE.equals(type)) {
-      convertRace(obs, dto);
+    CharacteristicCode type = null;
+    try {
+      type = CharacteristicCode.fromCode(obs.getCode()
+          .getCodingFirstRep()
+          .getCode());
+      dto.setType(type);
+      //Method + detail
+      CodeableConcept method = obs.getMethod();
+      dto.setMethod(CharacteristicMethod.fromCode(method.getCodingFirstRep()
+          .getCode()));
+      dto.setMethodDetail(method.getText());
+      //Value + detail
+      if (obs.getValue() instanceof CodeableConcept) {
+        CodeableConcept value = (CodeableConcept) obs.getValue();
+        dto.setValue(new CodingDto(value.getCodingFirstRep()
+            .getCode(), value.getCodingFirstRep()
+            .getDisplay()));
+        dto.setValueDetail(value.getText());
+      } else if (obs.hasComponent() && CharacteristicCode.ETHNICITY.equals(type)) {
+        convertEthnicity(obs, dto);
+      } else if (obs.hasComponent() && CharacteristicCode.RACE.equals(type)) {
+        convertRace(obs, dto);
+      }
+    } catch (FHIRException exc) {
+      dto.getErrors()
```

Comment:
if CharacteristicCode parsing fails - you will not set code and any other field. This is wrong